gef.py

```diff
@@ -10737,8 +10737,24 @@ def read_cstring(self,
         try:
             res_bytes = self.read(address, length)
         except gdb.error:
-            err(f"Can't read memory at '{address}'")
-            return ""
+            current_address = address
+            res_bytes = b""
+            while len(res_bytes) < length:
+                try:
+                    # Calculate how many bytes there are until next page
+                    next_page = current_address + DEFAULT_PAGE_SIZE
+                    page_mask = ~(DEFAULT_PAGE_SIZE - 1)
+                    size = (next_page & page_mask) - current_address
+
+                    # Read until the end of the current page
+                    res_bytes += self.read(current_address, size)
+
+                    current_address += size
+                except:
+                    if not res_bytes:
+                        err(f"Can't read memory at '{address}'")
```

Comment:
Address is an `int` here, but we should represent as hex


```suggestion
                        err(f"Can't read memory at '{address:#x}'")
```